### PR TITLE
Database Action Bug Fix

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Help Libray document is updated accordingly for Feature changes.

### CODE REVIEW

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs [hunk 1]
```diff
@@ -460,15 +462,14 @@ public void CloseConnection()
...
+        public async Task<List<string>> GetTablesListAsync(string Keyspace = null)
         {
 
             List<string> rc = new List<string>() { "" };
             if (MakeSureConnectionIsOpen())
             {
                 try
                 {
-                    //if (oConn == null || oConn.State == ConnectionState.Closed) Connect();
                     if (Database.DBType == Database.eDBTypes.Cassandra)
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Database/DatabaseOperations.cs:473]: Replace If with Switch

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs [hunk 2]
```diff
@@ -492,18 +493,30 @@ public List<string> GetTablesList(string Keyspace = null)
...
                     }
                     else
                     {
-                        DataTable table = oConn.GetSchema("Tables");
+                        DataTable table = await Task.Run(() => oConn.GetSchema("Tables"));
                         string tableName = "";
                         foreach (DataRow row in table.Rows)
                         {
                             switch (Database.DBType)
                             {
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Database/DatabaseOperations.cs:517]: Remove oracle case from switch as it is handled above
Also check if switch case is needed here as its being evealuated for each row in table.rows.

File: Ginger/GingerCoreNET/Database/DatabaseOperations.cs [hunk 3]
```diff
@@ -537,8 +550,13 @@ public List<string> GetTablesList(string Keyspace = null)
             }
             return rc;
         }
-
-
+        public string GetConnectedUsername()
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/Database/DatabaseOperations.cs:553]: Add comments that its being used for Oracle only

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

**Comment by AmanPrasad43** [comment]
Got some conflict

**Pull request closed without merging**